### ISSUE TITLE
Crash with Darktable jpeg

### ISSUE THREAD

**Issue by franckmee** [linuxmint/pix#140]
```
 * pix 2.6.5
 * Mint 20.2
```

**Issue**

Pix 2.6.5 crashes every time I try to load a JPEG produced by Darktable. Tested with different pictures generated from Darktable 3.0 to the current 3.6.

**Steps to reproduce**

Load a JPEG produced by Darktable, or a folder containing any such picture.
Alternately, display a folder in Pix, go to Darktable and export a JPEG to this folder. Pix will immediately crash.

**Expected behaviour**

Load and display the picture/folder.

**Other information**

Everything worked perfectly fine with the same files a few weeks ago (maybe before upgrading to Mint 20.2).

Example of picture that makes Pix crash:
![IMGP8190_01](https://user-images.githubusercontent.com/38156523/130596005-97a5f4f5-db30-46c3-a9ac-7fbf1e6e4bd1.jpg)

Text trace in terminal (note the index is always 19, no matter how many and what files are in the folder):
(pix:25297): GLib-GObject-WARNING **: 11:45:36.580: g_type_class_add_private() called multiple times for the same type
terminate called after throwing an instance of 'std::out_of_range'
  what():  basic_string::at: __n (which is 19) >= this->size() (which is 19)
Abandon (core dumped)

Crash file:
[_usr_bin_pix.1000.crash.gz](https://github.com/linuxmint/pix/files/7038318/_usr_bin_pix.1000.crash.gz)

**Comment by toastedsocks** [comment]
Ditto-  pix crashes out when trying to load a .jpg, .tiff or .png. Also happened to me after upgrading to Mint 20.2, something to do with the upgrade perhaps? As a test, I exported a .png from darktable, this crashed pix too, yet images exported pre-upgrade open up fine.
Regards, T.
P.S. I can confirm this behaviour also occurs with photos of geese.

**Comment by darkshram** [comment]
Must be something Linuxmint specific. I tried to replicate in ALDOS (a Fedora fork) using the sample image provided. No crash. Just a harmless glib warning.

- pix-2.6.5
- glib2-2.66.8
- gtk3-3.24.30
- gdk-pixbuf2-2.42.6
- libjpeg-turbo-2.0.6

![Captura de pantalla_2021-08-24_11-54-13](https://user-images.githubusercontent.com/10327500/130658089-26626598-6845-4fc0-be95-5da8eea2e9dc.png)

**Comment by toastedsocks** [comment]
Cheers for checking that out, darkshram, I've posted this issue on the Mint forum to see if anyone there has any ideas.
Crash report here-  [https://pastebin.com/KZ80jY6f](url)

**Comment by icarter09** [comment]
I put up a PR that addresses this issue. I tested it out with the provided image in the ticket. I'm going to keep researching this issue to see if I can find a better solution to handle the given exception. If anyone keeps having issues, let me know and we'll step from there. Thanks.

**Comment by tim747** [comment]
This issue is NOT just limited to "darktable"?
The image causing pix to crash for me is a digital camera jpeg which has been cropped with GIMP 2.8.22. Apart from image size, the only difference from the original in the image properties is the addition of XMP Exif data.  Both images previously displayed with no crash.
I am of course assuming that this is the same issue.
Trace file ([https://termbin.com/hocvx](url))
cheers

**Comment by tim747** [comment]
Update: If I turn off 'EXIF,IPTC,XMP support' Extension, pix does NOT crash.

**Comment by franckmee** [comment]
> Update: If I turn off 'EXIF,IPTC,XMP support' Extension, pix does NOT crash.

Confirmed. De-activating this extension prevents crashes. Re-activating it makes it crash again.

**Comment by rgreen5** [comment]
LM 21.1 Cinnamon (4.8.6) / Linux kernel 5.4.0-86-generic

Just confirming tim747 and franckmee's observations; turning off 'EXIF,IPTC,XMP support' stops the crash.

**Comment by darkshram** [comment]
Again, seems to be a Linuxmint specific issue. Still, can't reproduce crash.

Pix showing sample image provided, with EXIF, IPTC, XMP extension support enabled. **No crash**.

![Captura de pantalla de 2021-10-01 20-43-26](https://user-images.githubusercontent.com/10327500/135700018-2179d103-8448-4157-8ba1-02ea4a812f36.png)

- OS: ALDOS 1.4.16
- Cinnamon 5.0.5
- Cinnamon-desktop 5.0.0
- pix-2.6.5 (without any patch)
- glib2-2.66.8
- gtk3-3.24.30
- gdk-pixbuf2-2.42.6
- libjpeg-turbo-2.1.1
- **exiv2-libs-0.27.4**

Installed pixbuf loaders:

- heif-pixbuf-loader-1.12.0
- avif-pixbuf-loader-0.9.2
- heif-pixbuf-loader-1.12.0
- webp-pixbuf-loader-0.0.3
- avif-pixbuf-loader-0.9.2
- xcf-pixbuf-loader-0.0.1
- jxl-pixbuf-loader-0.5
- jp2-pixbuf-loader-0.0.2

Relevant ~/.xsession-errors messages:
```
(pix:6965): GLib-GObject-WARNING **: 20:51:53.923: g_type_class_add_private() called multiple times for the same type
```

**Comment by darkshram** [comment]
I have checked the ChangeLog of exiv2 and searched  for solved issues related to size. There is a relevant [bug in exiv2 < 0.27](https://dev.exiv2.org/issues/1338):

```Exiv2 aborted (crashed) with "Error in `exiv2': free(): invalid next size (fast)"```

For [what I see](https://forums.linuxmint.com/viewtopic.php?f=47&t=357829), Linuxmint comes with exiv2 0.25, which has this bug. So, seems the bug is really in exiv2 (not a pix bug) and gets solved upgrading to exiv2 >= 0.27 (current version is 0.27.4).

**Comment by franckmee** [comment]
> I have checked the ChangeLog of exiv2 and searched for solved issues related to size. There is a relevant [bug in exiv2 < 0.27](https://dev.exiv2.org/issues/1338):
> 
> `` Exiv2 aborted (crashed) with "Error in `exiv2': free(): invalid next size (fast)" ``
> 
> For [what I see](https://forums.linuxmint.com/viewtopic.php?f=47&t=357829), some versions of Linuxmint comes with exiv2 0.25, which has this bug. So, seems the bug is really in exiv2 (not a pix bug) and gets solved upgrading to exiv2 >= 0.27 (current version is 0.27.4).

Mine (Mint 20.2 Uma) was using exiv2 0.27.2, and I still got the crash. So I searched a 0.27.4 package to force the update, and it actually solved the crash.

So it seems we need to update to **exiv2 0.27.4** (earlier 0.27.2 is not enough). This [PPA](https://launchpad.net/~savoury1/+archive/ubuntu/graphics)'s version solved my issue.

Thanks a lot to everyone who helped!
